packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1836,9 +1770,8 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             localOpMetadata,
             opMetadataInternal,
         );
-        if (this.isContainerMessageDirtyable(type, content)) {
-            this.updateDocumentDirtyState(true);
-        }
+
+        this.updateDocumentDirtyState(true);
```

Comment:
This will change calculation of isDirty property on container (when using container with agent scheduler). It will be dirty (after changes) in places where it was not dirty before. This means that occasionally hosts (like Office.com, Teams) will prevent user from navigating away from container and saying user is about to lose some changes, when user might not have made any changes.

The chance of that is low (as leader election is engaged when there is some change in doc), but could happen (agent scheduler was the only message that was sent right before disconnection happened).

In general, I think we have to figure out how to allow users to override behaviors of ContainerRuntime. I.e. APIs like makeContainerRuntimeWithAgentScheduler() should accept container runtime factory, similar to how PureDataObjectFactory accepts factory for FluidDataStoreRuntime. That will allow users to augment properly functionality of runtime.